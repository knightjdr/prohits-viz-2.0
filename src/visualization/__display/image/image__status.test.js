import React from 'react';
import { shallow } from 'enzyme';

import Status from './image__status';

describe('Image status', () => {
  describe('with error', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Status
          error
          loading={false}
        />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render error content', () => {
      expect(wrapper.find('.image__status_red').length).toBe(1);
    });
  });

  describe('when loading', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Status
          error={false}
          loading
        />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render error content', () => {
      expect(wrapper.find('.image__status_default').length).toBe(1);
    });
  });

  describe('otherwise', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Status
          error={false}
          loading={false}
        />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render error content', () => {
      expect(wrapper.find('.image__status_red').length).toBe(1);
    });
  });
});
