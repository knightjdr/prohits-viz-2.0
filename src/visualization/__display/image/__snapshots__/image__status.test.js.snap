// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image status otherwise should match snapshot 1`] = `
<div
  className="image__status"
>
  <div
    className="image__status-inner"
  >
    <FontAwesomeIcon$1
      border={false}
      className="image__status_red"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [],
            "f071",
            "M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
          ],
          "iconName": "exclamation-triangle",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
    <span>
      There was an error displaying this image.
    </span>
  </div>
</div>
`;

exports[`Image status when loading should match snapshot 1`] = `
<div
  className="image__status"
>
  <div
    className="image__status-inner"
  >
    <FontAwesomeIcon$1
      border={false}
      className="image__status_default"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f013",
            "M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z",
          ],
          "iconName": "cog",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={true}
      symbol={false}
      transform={null}
    />
    <span>
      loading...
    </span>
  </div>
</div>
`;

exports[`Image status with error should match snapshot 1`] = `
<div
  className="image__status"
>
  <div
    className="image__status-inner"
  >
    <FontAwesomeIcon$1
      border={false}
      className="image__status_red"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [],
            "f071",
            "M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
          ],
          "iconName": "exclamation-triangle",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
    <span>
      There was an error displaying this image.
    </span>
  </div>
</div>
`;
